README.md

```diff
@@ -1,2 +1,177 @@
+
+The example files above are transformed to:
+
+`styles.css`:
+```js
+const stylesheet = new CSSStyleSheet();
+stylesheet.replaceSync(`.foo {\n  color: red;\n}\n`);
+export default stylesheet;
+```
+
+`index.js`:
+```js
+import styles from './styles.css';
+document.adoptedStyleSheets.push(styles);
+```
+
+This is a transparent and spec compliant transformation. Like native CSS modules, there are no side-effects from loading the CSS, the CSS is parsed only once, and there's a single `CSSStyleSheet` instance shared among all importers.
+
+After the transformation, Rollup can bundle the transformed CSS module into the JavaScript bundle(s) as usual.
+
+### Potential downsides
+
+While the transform is simple and spec compliant, there are two (small) potential downsides to be aware of:
+
+1. The CSS file can't be dynamic. This is a natural consequence of bundling. With native CSS modules you could import a CSS file from a remote URL or from a file path who's contents could change. With bundling the file is fixed at build time.
+2. The exact same CSS file can't be shared across JS imports and HTML imports. This is also a consequence of bundling. With native CSS modules you can load the CSS file in many different ways: import into JavaScript, with a `<link rel=stylesheet>` HTML tag, or some other CSS loader. With bundling you will have to leave a copy of the CSS as a plain .css file in order to load other ways.
+3. The CSS content is parsed as both JavaScript and CSS. This gives the transform a small performance and memory overhead compared to native CSS modules. On projects that use a similar technique of embedding plain CSS strings into JavaScript and creating `CSSStyleSheet` objects, the overhead was measured to be  small, since parsing strings in JavaScript is very fast.
+4. Dynamic import isn't supported (yet).
+
+## When to use
+
+1. To use CSS modules in browsers that don't support them
+2. To use CSS modules in Chrome versions that don't support import attributes
+3. To bundle CSS modules to reduce network requests
+
+## Brower support for CSS modules
+
+Currently, no browser supports both CSS modules and import attributes, so a transform is needed to use them.
+
+#### Chrome and Chromium browsers
+
+CSS modules are supported in Chrome [since version 93](https://chromestatus.com/feature/5948572598009856), but V8 only supports the older `assert` import _assertions_ syntax, and not the newer standard-track `with` import _attributes_ specification.
+
+#### Safari
+
+Safari supports constructible stylesheets (`new CSSStyleSheet()`), `adoptedStyleSheets`, and import attributes, but not CSS modules yet.
+
+#### Firefox
+
+Firefox supports constructible stylesheets (`new CSSStyleSheet()`) and `adoptedStyleSheets`, but not import attributes.
+
+## Future work
```

Comment:
another potential future work could be option to minify css?

Reply:
Good idea!